mlflow/tracking/fluent.py

```diff
@@ -1329,19 +1333,30 @@ def print_auto_logged_info(r):
         "pytorch_lightning": pytorch.autolog,
     }
 
-    def get_autologging_params(autolog_fn):
+    def get_autologging_params(integration, autolog_fn):
         try:
             needed_params = list(inspect.signature(autolog_fn).parameters.keys())
-            return {k: v for k, v in locals_copy if k in needed_params}
+            param_spec = inspect.signature(autolog_fn).parameters
+            default_params = {param.name: param.default for param in param_spec.values()}
+            return {k: AutologgingConfigManager.get_config(integration, k, default_params[k]) for k in needed_params}
         except Exception:
             return {}
 
+    def set_mlflow_autolog_params_in_config(integration):
+        try:
+            needed_params = list(inspect.signature(autolog).parameters.keys())
+            [AutologgingConfigManager.set_mlflow_config(integration, k, v)
+             for k, v in locals_copy if k in needed_params]
+        except Exception:
+            pass
+
     def setup_autologging(module):
         try:
             autolog_fn = LIBRARY_TO_AUTOLOG_FN[module.__name__]
-            autologging_params = get_autologging_params(autolog_fn)
+            autologging_params = get_autologging_params(module.__name__, autolog_fn)
+            autologging_params["_mlflow_called"] = True
             autolog_fn(**autologging_params)
```

Comment:
Given the situation:
```
mlflow.autolog(disable=False)
import sklearn
mlflow.autolog(disable=True)
```
The first line would set up an import hook which would run `mlflow.sklearn.autolog()` with the current configs at the time.
`mlflow.sklearn.autolog` receives a call (with some arguments), so it ends up updating the per-integration configs in ConfigManager with the configs that existed at the time of line 1.
Of course, the per-integration config for sklearn should be empty because there is no explicit user-called `mlflow.sklearn.autolog()`. This flag exists to differentiate between an implicit `mlflow.autolog` hook-added call and an explicit user-called `mlflow.sklearn.autolog()`.